osu.Game.Modes.Mania/ManiaRuleset.cs

```diff
@@ -102,6 +100,8 @@ public override IEnumerable<Mod> GetModsFor(ModType type)
 
         public override FontAwesome Icon => FontAwesome.fa_osu_mania_o;
 
+        public override KeyCounter[] GameplayKeys => new KeyCounter[] { /* Todo: Should be keymod specific */ };

```

Comment:
also when a virtual method is *creating* objects, i'd prefer it was a CreateGameplayKeys method instead of a property (to make it very obvious that it is instantiating new objects each call).